app/Domain/Trips/Actions/UpdateAction.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Domain\Trips\Actions;
+
+use App\Domain\Trips\Models\Trip;
+
+final class UpdateAction
+{
+    public function execute(Trip $trip, array $tripUpdateData): bool
+    {
+        return $trip->update($tripUpdateData);
```

Comment:
Is there a particular reason we've used a different convention here compared to the action above, where the subject/model is passed in the constructor rather than as a method param?

Reply:
The reason ive done the update to use a constructor is because im letting laravel inject the model using DI. im trying to move away form using `Model::query()->save` and have things injected at all times if i can. I cant do this via the constructor here on the update as im actually bringing the action in via DI as well